components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/GroupDAOImpl.java

```diff
@@ -18,188 +18,179 @@
+            for (String id : ids) {
+                try {
+                    groupIdsToNames.put(id, userStoreManager.getGroupNameByGroupId(id));
+                } catch (UserStoreClientException e) {
+                    // This is to ensure backward compatibility with the previous implementation.
+                    if (StringUtils.isNotBlank(e.getErrorCode()) && ERROR_NO_GROUP_FOUND_WITH_ID.getCode()
+                            .equals(e.getErrorCode())) {
+                        LOG.debug(String.format("No group found for the given group ID: %s in the tenantDomain: %s. " +
+                                "Therefore, skipping group id lookup.", id, tenantDomain));
+                    } else {
+                        throw e;
                     }
                 }
             }
+            return groupIdsToNames;
+        } catch (UserStoreException e) {
+            String errorMessage = "Error while resolving the group name for the given group Ids in " +
+                    "the tenantDomain: " + tenantDomain;
+            if (e instanceof UserStoreClientException) {
+                throw new IdentityRoleManagementClientException(INVALID_REQUEST.getCode(), errorMessage, e);
+            }
+            throw new IdentityRoleManagementServerException(UNEXPECTED_SERVER_ERROR.getCode(), errorMessage, e);
         }
-        return groupIdsToNames;
     }
 
     @Override
     public String getGroupIDByName(String name, String tenantDomain) throws IdentityRoleManagementException {
 
-        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        String groupID = null;
-        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
-            try (NamedPreparedStatement statement = new NamedPreparedStatement(connection, GET_GROUP_ID_BY_NAME_SQL)) {
-                statement.setInt(RoleConstants.RoleTableColumns.TENANT_ID, tenantId);
-                statement.setString(RoleConstants.RoleTableColumns.ROLE_NAME, name);
-                statement.setString(RoleConstants.RoleTableColumns.ATTR_NAME, RoleConstants.ID_URI);
-                int count = 0;
-                try (ResultSet resultSet = statement.executeQuery()) {
-                    while (resultSet.next()) {
-                        // Handle multiple matching groups.
-                        count++;
-                        if (count > 1) {
-                            String errorMessage =
-                                    "Invalid scenario. Multiple groups found for the given group name: " + name + " "
-                                            + "and tenantDomain: " + tenantDomain;
-                            throw new IdentityRoleManagementClientException(INVALID_REQUEST.getCode(), errorMessage);
-                        }
-                        groupID = resultSet.getString(1);
-                    }
+        String domainAwareName = UserCoreUtil.addDomainToName(name, tenantDomain);
```

Comment:
Why do we use the tenant domain as the domain for the group? AFAIK, we had domains for groups like PRIMARY, DEFAULT (don't know the exact context here).

Reply:
This is a bug.